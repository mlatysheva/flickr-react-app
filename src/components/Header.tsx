import { NavLink, useLocation } from 'react-router-dom';
import { ThemeSelector } from './Theme';

const CurrentPage = () => {
  const location = useLocation();
  const getCurrentPage = () => {
    switch (location.pathname) {
      case '/':
        return 'Home';
      case '/form':
        return 'Form';
      case '/about':
        return 'About';
      case '/api':
        return 'API';
      case '/error':
        return 'Error';
      default:
        return 'Error';
    }
  };

  return (
    <div className="current-page-title">
      <span className="here-text">Current page: </span>
      {getCurrentPage()}
    </div>
  );
};

function Header() {
  return (
    <header className="header">
      <CurrentPage />
      <nav className="nav">
        <NavLink to="/">Home</NavLink>
        <NavLink to="form">Form</NavLink>
        <NavLink to="api">API</NavLink>
        <NavLink to="about">About</NavLink>
      </nav>
      <ThemeSelector />
    </header>
  );
}

export default Header;
