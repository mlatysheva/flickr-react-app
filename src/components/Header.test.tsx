import { render, screen } from '@testing-library/react';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';

describe("Header", () => {
  it("renders About component", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>);
    expect(screen.getByText(/About/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  })
})