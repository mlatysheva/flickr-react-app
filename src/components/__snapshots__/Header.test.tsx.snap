// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders About component 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="current-page-title"
    >
      <span
        class="here-text"
      >
        Current page: 
      </span>
      Home
    </div>
    <nav
      class="nav"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        class=""
        href="/form"
      >
        Form
      </a>
      <a
        class=""
        href="/api"
      >
        API
      </a>
      <a
        class=""
        href="/about"
      >
        About
      </a>
    </nav>
    <div
      class="theme-selector theme-dark"
    />
  </header>
</DocumentFragment>
`;
